os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -206,37 +206,132 @@ binmgr_uinfo_t *binary_manager_get_udata(uint32_t bin_idx)
 }
 
 /****************************************************************************
- * Name: binary_manager_register_ubin
+ * Name: binary_manager_register_upart
  *
  * Description:
- *	 This function registers user binaries.
+ *	 This function registers a partition of user binaries.
  *
  ****************************************************************************/
-int binary_manager_register_ubin(char *name, uint32_t version, uint8_t load_priority)
+void binary_manager_register_upart(char *name, int part_num, int part_size)
 {
-	if (name == NULL || g_bin_count >= USER_BIN_COUNT) {
-		bmdbg("ERROR: Invalid parameter\n");
-		return ERROR;
+	int bin_idx;
+
+	if (part_num < 0 || part_size <= 0 || g_bin_count >= USER_BIN_COUNT) {
+		bmdbg("ERROR: Invalid part info : num %d, size %d\n", part_num, part_size);
+		return;
+	}
+
+	for (bin_idx = 1; bin_idx <= g_bin_count; bin_idx++) {
+		/* Already Registered */
+		if (!strncmp(BIN_NAME(bin_idx), name, strlen(name) + 1)) {
+			BIN_PARTNUM(bin_idx, 1) = part_num;
+			BIN_PARTSIZE(bin_idx, 1) = part_size;
+			bmvdbg("[USER%d : 2] %s size %d num %d\n", bin_idx, BIN_NAME(bin_idx), BIN_PARTSIZE(bin_idx, 1), BIN_PARTNUM(bin_idx, 1));
+			return;
+		}
 	}
 
 	/* Initialize binary table and register binary name, version and priority */
 	g_bin_count++;
 	BIN_ID(g_bin_count) = -1;
-	BIN_FILECNT(g_bin_count) = 1;
 	BIN_RTLIST(g_bin_count) = NULL;
 	BIN_NRTLIST(g_bin_count) = NULL;
 	BIN_STATE(g_bin_count) = BINARY_INACTIVE;
-	BIN_VER(g_bin_count, 0) = version;
-	BIN_LOAD_PRIORITY(g_bin_count, 0) = load_priority;
+	BIN_PARTNUM(g_bin_count, 0) = part_num;
+	BIN_PARTSIZE(g_bin_count, 0) = part_size;
 	strncpy(BIN_NAME(g_bin_count), name, BIN_NAME_MAX - 1);
 	BIN_NAME(g_bin_count)[BIN_NAME_MAX - 1] = '\0';
 	sq_init(&BIN_CBLIST(g_bin_count));
 
-	bmvdbg("[USER %d] %s\n", g_bin_count, BIN_NAME(g_bin_count));
+	bmvdbg("[USER%d : 1] %s size %d num %d\n", g_bin_count, BIN_NAME(g_bin_count), BIN_PARTSIZE(g_bin_count, 0), BIN_PARTNUM(g_bin_count, 0));
+}
 
-	return g_bin_count;
+/****************************************************************************
+ * Name: binary_manager_scan_ubin
+ *
+ * Description:
+ *	 This function scans user binaries and update information.
+ *
+ ****************************************************************************/
+bool binary_manager_scan_ubin(int bin_idx, bool crc_check)
+{
+	int ret;
+	int part_idx;
+	int latest_idx;
+	int bin_count;
+	uint32_t version;
+	uint32_t latest_ver;
+	char devpath[BINARY_PATH_LEN];
+	user_binary_header_t header_data[PARTS_PER_BIN];
+
+	if (bin_idx < 0 || bin_idx > binary_manager_get_ucount()) {
+		bmdbg("Invalid bin idx %d\n", bin_idx);
+		return ERROR;
```

Comment:
The return type of this function is bool.
But all the returns try to return `OK` or `ERROR`.
Please modify them or the return type.

Reply:
I set the return type to bool by mistake. I modified to `int`